### PR TITLE
[aznamespaces] Event Grid, now using full TypeSpec generation!

### PR DESCRIPTION
**Pull request by richardpark-msft**
The first generation of this client using pure TypeSpec, without swagger or autorest.

Some changes, in response to UX studies:
- Split the single client into two clients - SenderClient and ReceiverClient. This mirrors the structure we've used in other messaging libraries, and helps declutter each client.
- Remove 'CloudEvent' from all the method names, so ReceiveCloudEvents is now just Receive()
- topic and subscription are no longer method parameters, and are client level parameters instead.

TokenCredential auth is built-in, but testing is currently just using SAS keys due to #22961

### CODE REVIEW

File: sdk/messaging/eventgrid/aznamespaces/models.go [hunk 1]
```diff
@@ -53,6 +49,10 @@ type FailedLockToken struct {
 	LockToken *string
 }
 
+// publishResult - The result of the Publish operation.
+type publishResult struct {
```
Comment (jhendrixMSFT) [inline sdk/messaging/eventgrid/aznamespaces/models.go:53]: Is this something that's being generated?

Comment (richardpark-msft) [reply sdk/messaging/eventgrid/aznamespaces/models.go:53]: Yes.

Comment (jhendrixMSFT) [reply sdk/messaging/eventgrid/aznamespaces/models.go:53]: Mostly I was wondering if this was a base type that the emitter should have pruned. But that doesn't appear to be the case. It's harmless enough especially as it's not exported, but it does seem a bit weird. Might be worth following up with the event grid folks on why it's authored like this and if there's a better pattern.

Comment (richardpark-msft) [reply sdk/messaging/eventgrid/aznamespaces/models.go:53]: So I think I can just not use it at all (it is embedded in public responses, which is a big no-no). 

This all hangs on what they said in the last arch-board meeting about publish operations _never_ returning a result. So my instinct is to remove this, and change the signature of Send and SendEvents to only return only an 'error'. What do you think?

Comment (jhendrixMSFT) [reply sdk/messaging/eventgrid/aznamespaces/models.go:53]: I see. I think we should continue to return the response envelopes, just omit the `publishResult` type from them. This is consistent with our SDK patterns and is extensible in a non-breaking way.

Comment (richardpark-msft) [reply sdk/messaging/eventgrid/aznamespaces/models.go]: Brought it all back, we'll leave it as generated (except publishResult will be exported)

File: sdk/messaging/eventgrid/aznamespaces/CHANGELOG.md [hunk 2]
```diff
@@ -4,10 +4,15 @@
 
 ### Features Added
 
-- Added in "binary mode" support, which can improve efficiency when sending a single event (PR#22582)
+- Supports API version `2024-06-01`, including support for release with delay.
+- Added in binary mode support, which can improve efficiency when sending a single event (PR#22582)
```
Comment (jhendrixMSFT) [inline sdk/messaging/eventgrid/aznamespaces/CHANGELOG.md:8]: Is binary mode still available or did that get removed in a previous commit?

Comment (richardpark-msft) [reply sdk/messaging/eventgrid/aznamespaces/CHANGELOG.md]: It's gone now, didn't mean to leave the changelog entry there.

**Review by jhendrixMSFT — APPROVED** [review]

**Pull request merged**